marriage.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+function marriage() {
+    global $system;
+
+    global $player;
+    global $self_link;
+
+    if(isset($_POST['propose'])) {
+        try {
+            $to_marry = $system->clean($_POST['user_name']);
+
+            $result = $system->query("SELECT `user_id`, `user_name`, `spouse` FROM `users` WHERE `user_name`='{$to_marry}' LIMIT 1");
+
+            if(!$system->db_last_num_rows) {
+                throw new Exception("Invalid user!");
+            }
+
+            $user_to_marry = $system->db_fetch($result);
+
```

Comment:
Let's add a blocklist check here so players can't propose to someone who has them blocked